### ISSUE TITLE
NetHttpBinding with transport security not working

### ISSUE THREAD

**Issue by gvkries** [CoreWCF/CoreWCF#565]
I'm trying to port an WCF service that uses NetHttpBinding with `BasicHttpSecurityMode.Transport` and `HttpClientCredentialType.Certificate`. But this is not implemented, right? It looks like the message security is not populated in the `ServerWebSocketTransportDuplexSessionChannel::ProcessAuthentication()`method.

**Comment by mconnew** [comment]
I'll take a look.

**Comment by gvkries** [comment]
Just to clarify, `ServiceSecurityContext.Current` returns `null` in the case described above.

**Comment by mconnew** [comment]
Oh, I understand. That's expected behavior at the moment. You can get the client certificate from the HttpContext. We have a DI feature which allows you to add the HttpContext to your implementation method (you don't need to change anything in your contract). You can do the following:
```c#
[ServiceContract]
public interface IService
{
    string Echo(string message)
}

public partial class Service : IService
{
    public string Echo(string message, [CoreWCF.Injected] HttpContext httpContext)
    {
        var certificate = httpContext.Connection.ClientCertificate;
        // Do something with the certificate
        return message;
    }
}
````

The important thing to use this is to make your implementation partial so that a build time tool can back fill the method which matches your service contract interface. Will this work for you?

**Comment by gvkries** [comment]
Yes, that will work for me. Thank you very much.

**Comment by mconnew** [comment]
FYI in case it wasn't clear, you still need to configure asp.net core to require the client certificate as we don't have code to either enforce or validate that. I don't know yet how much control we have over configuring the client certificate requirement from CoreWCF (although I'm pretty sure we can't as CoreWCF gets involved too late). I think the best we can do is fail startup if it's misconfigured with a useful error message, but again I'm not sure how easy that is to do.

**Comment by gvkries** [comment]
Yes, that makes sense and I have it working correctly. Thanks again. 

But IMHO this should this be fixed for 1.0 anyway.
